Packs/Jira/Integrations/JiraAuditRecords/JiraAuditRecords.py

```diff
@@ -0,0 +1,220 @@
+
+        for logs in self._iter_events():
+            stored.extend(logs)
+
+            if len(stored) > max_fetch:
+                last_run['offset'] = last_run.get('offset', 0) + max_fetch
+                demisto.setLastRun(last_run)
+                return stored[:max_fetch]
+
+        last_run['offset'] = 0
+        demisto.setLastRun(last_run)
+        return stored
+
+    @staticmethod
+    def events_to_incidents(events: list):
+        incidents = []
+
+        for event in events:
+            incident = {
+                'name': f"Jira Audit Records - {event['summary']} - {event['id']}",
+                'occurred': event.get('created', '').removesuffix('+0000') + 'Z',
+                'rawJSON': json.dumps(event)
+            }
+            incidents.append(incident)
+
+        demisto.incidents(incidents)
+
+    @staticmethod
+    def set_next_run(log: dict) -> dict:
+        """
+            Handles and saves the values required for next fetch.
+
+            There are 3 values:
+                * from: From which time to fetch
+                * next_run: Time of creation of the last event fetched
+                * offset: The size of the offset (how many events to skip)
+
+            Since the rest API returns the events in desc order (the last event returns first), We need to save the last
+            event time creation in some variable (next_run) for the next fetches, in addition we need to save in another
+            variable (offset) the number of how many events we already fetched to skip them in the next fetch to avoid
+            duplicates, in addition we need to save the time (from) from when to fetch if there is still some incident
+            to fetch with offset
+        """
+        last_run = demisto.getLastRun()
+
+        if not last_run.get('next_time'):
+            last_time = log.get('created', '').removesuffix('+0000')
+            next_time = last_time[:-1] + str(int(last_time[-1]) + 1)
+
+            if last_run.get('offset'):
+                last_run['next_time'] = next_time
+            else:
+                last_run['from'] = next_time
+
+        else:
+            if not last_run.get('offset'):
+                last_run['from'] = last_run.pop('next_time')
+
+        return last_run
+
+
+def main():
+    # Args is always stronger. Get last run even stronger
+    demisto_params = demisto.params() | demisto.args() | demisto.getLastRun()
+
+    demisto_params['params'] = ReqParams.parse_obj(demisto_params)
+
+    request = Request.parse_obj(demisto_params)
+    client = Client(request)
+    get_events = GetEvents(client)
+    command = demisto.command()
+
+    if command == 'test-module':
+        get_events.run(max_fetch=1)
+        demisto.results('ok')
+
+    elif command in ('fetch-events', 'JiraAuditRecords-get-events'):
+        events = get_events.run(int(demisto_params.get('max_fetch', 100)))
+
+        if events:
+            demisto.setLastRun(get_events.set_next_run(events[0]))
```

Comment:
Are the events sorted?

Reply:
Yes
https://developer.atlassian.com/cloud/jira/platform/rest/v3/intro/#:~:text=Responses%20are%20listed%20in%20ascending%20order%20by%20default